OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -44,6 +44,10 @@ public class RefineryInfo : TraitInfo, Requires<WithSpriteBodyInfo>, IAcceptReso
 		[Desc("Discard resources once silo capacity has been reached.")]
 		public readonly bool DiscardExcessResources = false;
 
+		[GrantedConditionReference]
+		[Desc("Condition to grant while an actor is docked.")]
+		public readonly string DockedCondition = null;
```

Comment:
Is this sufficiently general to cover all use cases? `Cargo` allows for different passengers to grant different conditions.

Reply:
Hmm, I suppose we could make it more optional. Just not sure how. By docked actor name or by dock type.

In this PR there're no dock types just yet since you can only dock for unloading resources to refinery

Reply:
Docked actor name should be sufficient.

Reply:
It looks like I'll need to do a slight rething of DockClientManager to add per DockType conditions